xdas/compose.py

```diff
@@ -0,0 +1,530 @@
+from collections import Counter, UserDict
+from collections.abc import Callable, Hashable
+from copy import copy
+from typing import Any, Self, Type
+
+from .database import Database
+from .processing import DatabaseLoader, DatabaseWriter, ProcessingChain
+
+ChainType = Type[ProcessingChain]
+LoaderType = Type[DatabaseLoader]
+WriterType = Type[DatabaseWriter]
+DatabaseType = Type[Database]
+
+
+class Sequence(UserDict):
```

Comment:
`OPSequence`? `OP` indicating operations. I thought you wanted to call it `DataCollection` and not `DataSequence` (which would imply some temporal trajectory, which is not necessarily the case with multiple instruments)

Reply:
So DataCollection comes into two flavours: DataSequence and DataMapping.

Reply:
I think it would be beneficial to keep the naming convention of public classes/methods as light as possible. I can imagine that casual users would more likely engage with `Atom` and `Sequence` than with `DataCollection`, which seems like a much more of an advanced feature at the level of a network operator. In the case of ABYSS, Marie would be in charge of dealing with the `DataCollection`, and the rest of us (esp. students) would only be using `Sequence` that acts on the outcome of `DataCollection`. In that case, it would not be helpful to refactor `Sequence` into the more obscure `OPSequence`.